nighthawk/test/client_test.cc

```diff
@@ -0,0 +1,147 @@
+        "nighthawk/test/test_data/benchmark_http_client_test_envoy.yaml"));
+    envoy_config = Envoy::TestEnvironment::substitute(envoy_config);
+  }
+
+  void SetUp() override {
+    // We fork the integration test fixture into a child process, to avoid conflicting
+    // runtimeloaders as both NH and the integration server want to own that and we can have only
+    // one. The plan is to move to python for this type of testing, so hopefully we can deprecate
+    // this test and it's peculiar setup with fork/pipe soon.
+    pipe(fd_port);
+    pipe(fd_confirm);
+    pid_ = fork();
+    RELEASE_ASSERT(pid_ >= 0, "Fork failed");
+
+    const int kParentMessageId = 12341234;
+
+    if (pid_ == 0) {
+      // child process running the integration test server.
+      ares_library_init(ARES_LIB_INIT_ALL);
+      Envoy::Event::Libevent::Global::initialize();
+      initialize();
+      int port = lookupPort("listener_0");
+      int parent_message;
+      write(fd_port[1], &port, sizeof(port));
+      // The parent process writes to fd_confirm when it has read the port. This call to read blocks
+      // until that happens.
+      read(fd_confirm[0], &parent_message, sizeof(parent_message));
+      RELEASE_ASSERT(parent_message == kParentMessageId, "Unexpected kParentMessageId value");
+      // The parent process closes fd_port when the test tears down. The read call blocks until it
+      // does that.
+      RELEASE_ASSERT(read(fd_port[0], &port_, sizeof(port_)) == -1, "read failed");
+      GTEST_SKIP();
+    } else if (pid_ > 0) {
+      RELEASE_ASSERT(read(fd_port[0], &port_, sizeof(port_)) > 0, "read failed");
+      RELEASE_ASSERT(port_ > 0, "read unexpected port_ value");
+      RELEASE_ASSERT(write(fd_confirm[1], &kParentMessageId, sizeof(kParentMessageId)) ==
+                         sizeof(kParentMessageId),
+                     "write failed");
+    }
+  }
+
+  void TearDown() override {
+    if (pid_ == 0) {
+      test_server_.reset();
+      fake_upstreams_.clear();
+      ares_library_cleanup();
+    }
+    RELEASE_ASSERT(!close(fd_confirm[0]), "close failed");
+    RELEASE_ASSERT(!close(fd_confirm[1]), "close failed");
+    RELEASE_ASSERT(!close(fd_port[0]), "close failed");
+    RELEASE_ASSERT(!close(fd_port[1]), "close failed");
+  }
+
+  std::string testUrl() {
+    RELEASE_ASSERT(pid_ > 0, "Unexpected call to testUrl");
+    const std::string address = Envoy::Network::Test::getLoopbackAddressUrlString(GetParam());
+    return fmt::format("http://{}:{}/", address, port_);
+  }
+
+  std::unique_ptr<OptionsImpl> createOptionsImpl(const std::string& args) {
+    std::vector<std::string> words = Envoy::TestUtility::split(args, ' ');
+    std::vector<const char*> argv;
+    for (const std::string& s : words) {
+      argv.push_back(s.c_str());
+    }
+    return std::make_unique<OptionsImpl>(argv.size(), argv.data());
+  }
+
+  int port_;
+  pid_t pid_;
+  int fd_port[2];
+  int fd_confirm[2];
+  static std::string envoy_config;
```

Comment:
Static non-PoD disallowed due to static initialization fiasco by the style guide, consider making this non-static, using const char[] or `CONSTRUCT_ON_FIRST_USE` instead.